### PR TITLE
Backport node10

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** PR backports code already pulled into master from https://github.com/nasa/cumulus/pull/1393

Addresses [CUMULUS-1626](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1626)

## Changes

- **CUMULUS-1626**
  - Updates Cumulus to use node10/CMA 1.1.2 for all of it's internal lambdas in prep for AWS node 8 EOL

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

**Comment by Jkovarik** [comment]
This PR updates the following: 

* All references to node 8.10 to the same version of node code is running in for our AWS Lambdas. 
* AWS SDK to 2.585.0
* Updates the es indexer to handle extra fields in AWS Linux 2's logs, adds a logic-specific unit test around that logic
* Updates all references to cumulus-message-adapter.js to utilize v1.1.0
* Updates the CMA image target to a layer that uses v1.1.2 
* Updates JSFTP to utilize a fork with multiple fixes related to node10/etc (there is a ticket to move fully from this library) 
* Updates libraries that were throwing test-failing  deprecation warnings under node 10.

File: packages/api/ecs/async-operation/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:8.10
+FROM node:10.16.3
```
Comment (yjpa7145) [inline packages/api/ecs/async-operation/Dockerfile]: We'll need to release an updated version of the `async-operation` image to Docker Hub.

Comment (Jkovarik) [reply packages/api/ecs/async-operation/Dockerfile]: Done, and updated 4095166

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Updated
```
Comment (markdboyd) [inline CHANGELOG.md]: Is this needed in the backport?

File: example/package.json [hunk 3]
```diff
@@ -55,11 +55,12 @@
...
     "@cumulus/sync-granule": "1.14.4",
     "@cumulus/test-processing": "1.14.4",
-    "aws-sdk": "^2.227.1",
+    "aws-sdk": "^2.585.0",
     "child-process-promise": "^2.2.1",
     "lodash.differencewith": "^4.5.0",
     "lodash.isequal": "^4.5.0",
-    "p-retry": "^2.0.0"
+    "p-retry": "^2.0.0",
+    "ssh2-streams": "^0.4.8"
```
Comment (markdboyd) [inline example/package.json]: Where is this package used?

File: packages/api/es/indexer.js [hunk 4]
```diff
@@ -39,6 +39,50 @@ async function createIndex(esClient, indexName) {
   log.info(`Created esIndex ${indexName}`);
 }
 
+
+/**
+ * Parses a StepFunction log payload  and returns a es logsrecord object
```
Comment (markdboyd) [inline packages/api/es/indexer.js]:
```suggestion
 * Parses a StepFunction log payload and returns an Elasticsearch log record object
```

File: example/package.json (hunk 3 above)
Comment (Jkovarik) [reply example/package.json]: The package has a sub dependency that is too permissive and un-upgradeable.  ssh2-streams sub .... .4? results in a buffer constructor deprecation warning under node 10 and later.

File: CHANGELOG.md (hunk 2 above)
Comment (Jkovarik) [reply CHANGELOG.md]: yes, as we update the changelog for backport releases... although that's non-linear so perhaps we should re-evaluate the formatting going forward.

**Pull request merged**